server/src/main/java/org/elasticsearch/common/util/concurrent/ReleasableLock.java

```diff
@@ -32,7 +32,7 @@
     private final Lock lock;
 
     /* a per thread boolean indicating the lock is held by it. only works when assertions are enabled */
```

Comment:
can you update this comment to reflect the new behavior?

Reply:
I pushed 14593388a2446b6bf63753092e000f56728db1dd.